### PR TITLE
Pchr 1315 contact summary is blank at first load

### PR DESCRIPTION
**Pull request by danilo-compucorp**
**Issue:** When you go to a contact's page on http://civihr-1.6-dev.civihrhosting.co.uk, you get blank Contact Summary. If I go to other page, and come back to this contact page by hitting browser's back button, then the Contact Summary loads.

The issue seems to be caused by a race condition between the components.
Other than solving the reported issue, I did see some other issues in the browser's console. The first one was related to the absence on the "momentjs" definition. After fixing this issue, a second one appeared: 
> Cannot read property 'length' of undefined

Which was being caused by undefined _response.values_, and I was able to track down to a malformed SQL query being executed in the server.  Basically, the query was *always* expecting some "absence type ids" to be present. After fixing this issue, it's apparently working fine (at least on my local environment). Also, this PR fixes the existing eslint configuration file, and adjusts some coding styles.

### CODE REVIEW

**Comment by AkA84** [comment]
The issue described in this PR doesn't seem to be about the issue described in PCHR-1315:

> When you go to a contact's page on http://civihr-1.6-dev.civihrhosting.co.uk, you get blank Contact Summary.

**Comment by AkA84** [comment]
> Although I was unable to reproduce the specified error

So you're not 100% sure the fixes in this PR are gonna address the issue described in the ticket?

> After fixing this issue, it's apparently working fine

Working fine in respect to what? To getting some console errors? In your local you could see the tab content fine, right?

File: contactsummary/js/dist/contact-summary.min.js [hunk 1]
```diff
@@ -1 +1,1763 @@
..."contact-summary/controllers/leave","contact-summary/controllers/sickness","contact-summary/directives/donutChart"],function(e){var t=e.module("contactsummary",["ngRoute","ngResource","ui.bootstrap","contactsummary.controllers","contactsummary.directives","contactsummary.filters","contactsummary.services","contactsummary.settings"]);t.config(["settings","$routeProvider","$resourceProvider","$httpProvider","$logProvider",function(e,t,n,r,i){i.debugEnabled(e.debug),t.when("/",{controller:"ContactSummaryCtrl",controllerAs:"ContactSummaryCtrl",templateUrl:e.pathBaseUrl+e.pathTpl+"mainTemplate.html",resolve:{}}).otherwise({redirectTo:"/"}),n.defaults.stripTrailingSlashes=!1,r.defaults.headers.common["X-Requested-With"]="XMLHttpRequest"}]),t.run(["settings","$rootScope","$q","$log",function(e,t,n,r){r.debug("app.run"),t.pathTpl=e.pathTpl,t.prefix=e.classNamePrefix}])}),require.config({urlArgs:"bust="+(new Date).getTime(),paths:{"contact-summary":CRM.vars.contactsummary.baseURL+"/js/src/contact-summary"}}),require(["contact-summary/app"],function(){document.addEventListener("contactsummaryLoad",function(){angular.bootstrap(document.getElementById("contactsummary"),["contactsummary"])})});
```
Comment (AkA84) [inline contactsummary/js/dist/contact-summary.min.js]: I don't know why, but now this minified file is not compressed anymore

File: contactsummary/CRM/Contactsummary/Utils/Absences.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ private static function getAbsenceDuration($absenceTypeNames = array(), $periodI
       ON (a.id = a1.source_record_id AND a1.activity_type_id = %1 AND a1.status_id = %2)
 
       WHERE
-        a.id IN (" . implode(',', $absenceTypeIds) . ")
+        a.id IN (" . (count($absenceTypeIds) > 0 ? implode(',', $absenceTypeIds) : 'NULL') . ")
```
Comment (AkA84) [inline contactsummary/CRM/Contactsummary/Utils/Absences.php]: How did you get this line to throw an error? You didn't have any absence type defined?

Comment (danilo-compucorp) [inline contactsummary/CRM/Contactsummary/Utils/Absences.php]: Apparently not.

File: contactsummary/js/dist/contact-summary.min.js (hunk 1 above)
Comment (danilo-compucorp) [inline contactsummary/js/dist/contact-summary.min.js]: Yeah, it turns out that the file _contactsummary/js/src/contact-summary/controllers/contactSummary.js_ was missing an empty line at the end. This is also fixed now

**Comment by AkA84** [comment]
Approved

**Pull request merged**